src/components/useRequest.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2020 RoadieHQ
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { useAsync } from 'react-use';
+import { Octokit } from '@octokit/rest';
+import { Entity } from '@backstage/catalog-model';
+import { useApi, githubAuthApiRef, configApiRef } from '@backstage/core';
+import { useProjectEntity } from './useProjectEntity';
+
+ export const useRequest = (
+     entity: Entity,
+     requestName: string,
+     perPage: number = 0,
+     maxResults: number = 0,
+     showTotal: boolean = false
+) => {
+    const auth = useApi(githubAuthApiRef);
+    const config = useApi(configApiRef);
+    const providerConfigs = config.getOptionalConfigArray('integrations.github') ?? [];
+    const targetProviderConfig = providerConfigs[0];
+    const baseUrl = targetProviderConfig?.getOptionalString('apiBaseUrl');
+    
+    const { owner, repo } = useProjectEntity(entity);
+    const { value, loading, error } = useAsync(async (): Promise<
+    any
+  > => {
+    const token = await auth.getAccessToken(['repo']);
+    const octokit = new Octokit({auth: token});
+    const response = await octokit.request(`GET /repos/{owner}/{repo}/${requestName}`, {
+      ...(baseUrl && {baseUrl}),
+      owner,
+      repo,
+      ...(perPage && {per_page: perPage}),
+    });
+    const data = await response.data;
```

Comment:
What is happening in this line? Is this accessing the `data` property of the response? How can that be an async operation. I would have expected it to be `const data = response.data` or `const data = await response.data()`.

Reply:
Perhaps there is just something about Typescript that I don't understand.

Reply:
Octokit.request return Promise but you are right second await is not needed here.